# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalChild.cpp
# Commit: ccd96b99aea2
# Full Hash: ccd96b99aea28e8214e585b864d3fb2245bac142
# Author: Atila Butkovits <abutkovits@mozilla.com>
# Date: 2025-06-07 08:31:23
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing failures complaining about fedcm-no-registered-idps.https.html.
#   
#   This reverts commit ba33457e0cbac24a8b7d963c45138c62e92a74ae.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 4bfadfc33207 -r ccd96b99aea2 dom/ipc/WindowGlobalChild.cpp
--- a/dom/ipc/WindowGlobalChild.cpp	Fri Jun 06 21:07:41 2025 +0300
+++ b/dom/ipc/WindowGlobalChild.cpp	Fri Jun 06 21:35:18 2025 +0300
@@ -575,6 +575,26 @@
   return IPC_OK();
 }
 
+IPCResult WindowGlobalChild::RecvNavigateForIdentityCredentialDiscovery(
+    const nsCString& aURI, const IdentityLoginTargetType& aType) {
+  AutoJSAPI jsapi;
+  if (!jsapi.Init(GetWindowGlobal())) {
+    return IPC_OK();
+  }
+  MOZ_ASSERT(WindowContext()->TopWindowContext());
+  nsGlobalWindowOuter* outer = nsGlobalWindowOuter::GetOuterWindowWithId(
+      WindowContext()->TopWindowContext()->OuterWindowId());
+  bool popup = aType == IdentityLoginTargetType::Popup;
+  RefPtr<dom::BrowsingContext> newBC;
+  if (popup) {
+    Unused << outer->OpenJS(aURI, u"_blank"_ns, u"popup"_ns,
+                            getter_AddRefs(newBC));
+    return IPC_OK();
+  }
+  Unused << outer->OpenJS(aURI, u"_top"_ns, u""_ns, getter_AddRefs(newBC));
+  return IPC_OK();
+}
+
 void WindowGlobalChild::SetDocumentURI(nsIURI* aDocumentURI) {
   // Registers a DOM Window with the profiler. It re-registers the same Inner
   // Window ID with different URIs because when a Browsing context is first